cclib/bridge/cclib2horton.py

```diff
@@ -0,0 +1,141 @@
+
+    if (hortonver == 2):
+        supportedAttrs = {"atomnos": "numbers", "mult": "ms2", "polarizability": "polar"}
+        inputattr = ccdat.__dict__
+    
+        attributes = dict((supportedAttrs[oldKey], val) for (oldKey, val) in inputattr.items() if (oldKey in supportedAttrs))
+    
+        # These need some separate care:
+        if hasattr(ccdat, "atomcoords"):
+            attributes["coordinates"] = ccdat.atomcoords[-1]
+        if hasattr(ccdat, "mocoeffs"):
+            attributes["orb_alpha"] = ccdat.mocoeffs[0]
+            # if unrestricted:
+            if len(ccdat.mocoeffs) == 2:
+                attributes["orb_beta"] = ccdat.mocoeffs[1]
+        if hasattr(ccdat, "coreelectrons") and isinstance(ccdat.coreelectrons, numpy.ndarray):
+            attributes["pseudo_numbers"] = ccdat.coreelectrons
+        if hasattr(ccdat, "atomcharges") and "mulliken" in ccdat.atomcharges:
+            attributes["mulliken_charges"] = ccdat.atomcharges["mulliken"]
+        if hasattr(ccdat, "atomcharges") and "natural" in ccdat.atomcharges:
+            attributes["npa_charges"] = ccdat.atomcharges["natural"]
+            
+    elif (hortonver == 3):
+        # In horton 3 IOData, inputs are type-checked
+        # if coordinates are known; numpy.ndarray (size natom-by-3)
+        if (hasattr(ccdat, 'atomcoords') and isinstance(ccdat.atomcoords, numpy.ndarray)):
+            attributes['atcoords'] = ccdat.atomcoords[-1]
+        # if atomic numbers are known; numpy.ndarray (size natom)
+        if (hasattr(ccdat, 'atomnos') and isinstance(ccdat.atomnos, numpy.ndarray)):
+            attributes['atnums'] = ccdat.atomnos,
+        # if orbital coefficients known; iodata.orbitals.MolecularOrbitals
+        if (hasattr(ccdat, 'mocoeffs') and isinstance(ccdat.mocoeffs, numpy.ndarray)):
+            attributes['mo'] = MolecularOrbitals(kind = 'restricted', norba = len(ccdat.mocoeffs[0]), coeffs = ccdat.mocoeffs[0].T)
+            # and if unrestricted:
+            if (len(mocoeffs) == 2):
+                attributes['mo'].kind = 'unrestricted'
+                attributes['mo'].norbb = len(ccdat.mocoeffs[1])
+                attributes['mo'].coeffs.append(ccdat.mocoeffs[1].T)
+        # if multiplicity known; float / should not be set when mocoeffs present
+        if (hasattr(ccdat, 'mult') and ccdat.mult != None and not isinstance(ccdat.mocoeffs, numpy.ndarray)):
+            attributes['spinpol'] = ccdat.mult - 1 # horton has 2S+1, iodata has 2S
+        # if pseudopotentials exist; numpy.ndarray (size natom)
+        if (hasattr(ccdat, 'coreelectrons') and isinstance(ccdat.coreelectrons, numpy.ndarray)):
+            attributes['atcorenums'] = ccdat.coreelectrons
+        # if mulliken charges are known; dict of numpy.ndarrays (size natom)
+        if (hasattr(ccdat, 'atomcharges') and isinstance(ccdat.atomcharges, dict)):
+            attributes['atcharges'] = ccdat.atomcharges
+
+    return IOData(**attributes)
+
+
+def makecclib(iodat):
```

Comment:
Another thing you could have done is put makehorton and makecclib in separate PRs.